controller/appcontroller.go

```diff
@@ -477,11 +477,22 @@ func (ctrl *ApplicationController) processAppOperationQueueItem() (processNext b
 	return
 }
 
+// shouldbeDeleted returns whether a given resource obj should be deleted on cascade delete of application app
 func shouldBeDeleted(app *appv1.Application, obj *unstructured.Unstructured) bool {
 	return !kube.IsCRD(obj) && !isSelfReferencedApp(app, kube.GetObjectRef(obj))
 }
 
-func (ctrl *ApplicationController) finalizeApplicationDeletion(app *appv1.Application) error {
+// Checks whether an object is permitted to be deleted according to the project restrictions
+func (ctrl *ApplicationController) isPermittedForDelete(app *appv1.Application, obj *unstructured.Unstructured) bool {
+	permitted := false
+	if proj, err := ctrl.getAppProj(app); err == nil {
```

Comment:
I think we should error here

Reply:
Agreed. It makes the function not inline expressable, but I guess it's worth it. Committed appropriate changes.